app/models/manageiq/providers/kubernetes/container_manager/event_catcher_mixin.rb

```diff
@@ -47,14 +47,15 @@ def monitor_events
     reset_event_monitor_handle
   end
 
-  def process_event(event)
+  def queue_event(event)
     event_data = extract_event_data(event)
-    if event_data.nil?
-      _log.debug "#{log_prefix} Discarding event [#{event_data}]"
-    else
-      _log.info "#{log_prefix} Queuing event [#{event_data}]"
-      EmsEvent.add_queue('add_kubernetes', @cfg[:ems_id], event_data)
-    end
+    _log.info "#{log_prefix} Queuing event [#{event_data}]"
+    EmsEvent.add_queue('add_kubernetes', @cfg[:ems_id], event_data)
+  end
+
+  def filtered?(event)
+    event_data = extract_event_data(event)
```

Comment:
I think the filtering logic can be different for different providers? In that case the current code looks more suitable where we have a different method to check if the event belongs to filtered category or not. The base runner method "process_event" uses queue_event and filtered? methods, so if we want to merge the methods, then the logic in queue_event has to contain filtering logic too for all providers.

About processing event data twice, I agree that may be the methods should get extracted event itself rather than the current event object. I will try to address it in the follow up PR.